src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,305 @@
+                .attribute(GATKSVVCFHeaderLines.ALIGN_LENGTHS, alignLengths.stream().map(String::valueOf).collect(Collectors.joining(",")))
+                .attribute(GATKSVVCFHeaderLines.MAX_ALIGN_LENGTH, maxAlignLength)
+                .attribute(GATKSVVCFHeaderLines.BREAKPOINT_IDS, breakpointIds.stream().collect(Collectors.joining(",")))
+                .attribute(GATKSVVCFHeaderLines.CONTIG_IDS, assembledContigIds.stream().map(s -> s.replace(" ", "_")).collect(Collectors.joining(",")));
+
+        if (breakpointAllele.insertedSequence.length() > 0) {
+            vcBuilder = vcBuilder.attribute(GATKSVVCFHeaderLines.INSERTED_SEQUENCE, breakpointAllele.insertedSequence);
+        }
+
+        if (breakpointAllele.homology.length() > 0) {
+            vcBuilder = vcBuilder.attribute(GATKSVVCFHeaderLines.HOMOLOGY, breakpointAllele.homology);
+            vcBuilder = vcBuilder.attribute(GATKSVVCFHeaderLines.HOMOLOGY_LENGTH, breakpointAllele.homology.length());
+        }
+
+        if (is5To3Inversion(breakpointAllele)) {
+            vcBuilder = vcBuilder.attribute(GATKSVVCFHeaderLines.INV_5_TO_3, "");
+        }
+
+        if (is3To5Inversion(breakpointAllele)) {
+            vcBuilder = vcBuilder.attribute(GATKSVVCFHeaderLines.INV_3_TO_5, "");
+        }
+
+
+        return vcBuilder.make();
+    }
+
+    private static boolean is3To5Inversion(final BreakpointAllele breakpointAllele) {
+        return ! breakpointAllele.fiveToThree && breakpointAllele.threeToFive;
+    }
+
+    private static boolean is5To3Inversion(final BreakpointAllele breakpointAllele) {
+        return breakpointAllele.fiveToThree && ! breakpointAllele.threeToFive;
+    }
+
+    private static String getInversionId(final BreakpointAllele breakpointAllele) {
+        final String invType;
+        if (is5To3Inversion(breakpointAllele)) {
+            invType = "INV_5_TO_3";
+        } else if (is3To5Inversion(breakpointAllele)) {
+            invType = "INV_3_TO_5";
+        } else {
+            // I don't think this should happen
+            invType = "INV_UNKNOWN";
+        }
+
+        return invType + "_" + breakpointAllele.leftAlignedLeftBreakpoint.getContig() + "_" + breakpointAllele.leftAlignedLeftBreakpoint.getStart() + "_" + breakpointAllele.leftAlignedRightBreakpoint.getStart();
+    }
+
+
+    public static Tuple2<BreakpointAllele, Tuple2<Tuple2<String,String>, AssembledBreakpoint>> keyByBreakpointAllele(final Tuple2<Tuple2<String,String>, AssembledBreakpoint> breakpointIdAndAssembledBreakpoint) {
+        final BreakpointAllele breakpointAllele = breakpointIdAndAssembledBreakpoint._2.getBreakpointAllele();
+        return new Tuple2<>(breakpointAllele, new Tuple2<>(breakpointIdAndAssembledBreakpoint._1, breakpointIdAndAssembledBreakpoint._2));
+    }
+
+    public static Boolean inversionBreakpointFilter(final Tuple2<BreakpointAllele, Tuple2<Tuple2<String,String>,AssembledBreakpoint>> breakpointAlleleTuple2Tuple2) {
```

Comment:
To me this method is better named as `inversionBreakpointAlleleFilter`, because `AssembledBreakpoint`'s are not filtered, rather it is the `BreakpointAllele`'s (i.e. keys of input).